### PR TITLE
update for golang 1.17

### PR DESCRIPTION
**Pull request by elmiko**
This change updates the go.mod and Dockerfiles to use golang 1.17, it
also updates the vendor directory. The vendor directory was updated by
running `go mod vendor && go mod tidy && go mod verify`.

### CODE REVIEW

**Comment by mandre** [comment]
Not sure if the openstack-proxy job is supposed to pass. Let's give it another try:
/test e2e-openstack-proxy

**Comment by elmiko** [comment]
update
* changed golang version in ci-operator.yaml file

**Comment by elmiko** [comment]
update
* ran `go fmt` on the repo to fix tag issues with 1.17

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
-FROM registry.ci.openshift.org/openshift/release:golang-1.15 AS builder
+FROM registry.ci.openshift.org/openshift/release:golang-1.17 AS builder
 WORKDIR /go/src/sigs.k8s.io/cluster-api-provider-openstack
 COPY . .
 
 RUN go build -o ./machine-controller-manager ./cmd/manager
 
-FROM registry.ci.openshift.org/origin/4.8:base
+FROM registry.ci.openshift.org/openshift/origin-v4.0:base
```
Comment (pierreprinetti) [inline Dockerfile:7]: `4.0`? Is this intentional, or was it meant to be `4.10`?

Comment (elmiko) [reply Dockerfile:7]: i tried to make this look like our other community-facing images, see the aws provider for example https://github.com/openshift/cluster-api-provider-aws/blob/master/Dockerfile#L8

my thinking here is that the "Dockerfile" files are for making publicly available images and the "Dockerfile.rhel" is for making the subscription-based images. so i was basically trying to make the community ones look similar, and afaik they all work with that 4.0 base.

Comment (pierreprinetti) [reply Dockerfile:7]: oh, ok.

**Pull request merged**